sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CollationExpressionSuite.scala

```diff
@@ -161,4 +162,40 @@ class CollationExpressionSuite extends SparkFunSuite with ExpressionEvalHelper {
       checkEvaluation(ArrayExcept(left, right), out)
     }
   }
+
+  test("MultiLikeBase regexp expressions with collated strings") {
```

Comment:
Why do we need this test if we already have checks for `LikeAll`, `LikeAny`, etc. in `CollationRegexpExpressionsSuite`?

Reply:
with StringExpressions, we had a situation where interpreted & codegen codepaths defaulted to a single function call, so unit tests in CollationSupportSuite were enough to verify everything goes well there

however here we have to make traditional unit tests in order to verify the behaviour for both interpreted (nullSafeEval) & codegen (doGenCode) execution (sql tests don't seem to cover both)

this PR will only be correct once I add unit tests for all other regexp expressions too, and I figured CollationExpressionSuite is the correct place to do it

Reply:
I am a bit sceptical this is the right place for the test. I agree we need to have it, but does it really belong with CollationExpressions? I would assume here we only expect tests that check behaviour of Collation and Collate expressions. Could we maybe move this to CollationRegexpExpressionSuite?